packages/eslint-plugin/src/rules/no-confusing-void-expression.ts

```diff
@@ -334,5 +340,24 @@ export default util.createRule<Options, MessageId>({
 
       return ['(', '[', '`'].includes(startToken.value);
     }
+
+    function canFix(node: TSESTree.Node): boolean {
+      const services = util.getParserServices(context);
+
+      let targetNode: TSESTree.Node | null = null;
+
+      if (node.type === AST_NODE_TYPES.ReturnStatement) {
+        targetNode = node.argument;
+      } else if (node.type === AST_NODE_TYPES.ArrowFunctionExpression) {
+        targetNode = node.body;
+      }
+
+      if (targetNode) {
+        const type = util.getConstrainedTypeAtLocation(services, targetNode);
+        return tsutils.isTypeFlagSet(type, ts.TypeFlags.VoidLike);
+      }
+
+      return true;
```

Comment:
[Cleanup] I tried removing this locally and all tests passed. Dead code?

This `canFix` function is only ever called with `TSESTree.ArrowFunctionExpression` or `TSESTree.ReturnStatement`. So I suspect the `if (targetNode) {` is unnecessary.

Reply:
Hi @JoshuaKGoldberg ,
When the node type is `TSESTree.ReturnStatement` it's `argument` property is nullable. 
However, the `argument` property will never actually be null as you told. Because this rule start to find ancestor node from one of `AwaitExpression, CallExpression, TaggedTemplateExpression`.
Maybe should i use `non-null-assertion` ?

Reply:
Yeah if you're 100% confident the type system is wrong, then an assertion would be the right strategy.